### PR TITLE
fix: subquery selects

### PR DESCRIPTION
**Pull request by barakalon**
We weren't propagating `selects` from subqueries

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -846,10 +846,14 @@ def alias_column_names(self):
 
     @property
     def selects(self):
+        # For EXPLODE
```
Comment (GeorgeSittas) [inline sqlglot/expressions.py:849]: What does this comment mean?

**Review by GeorgeSittas — APPROVED** [review]

File: sqlglot/expressions.py (hunk 1 above)
Comment (barakalon) [reply sqlglot/expressions.py:849]: From what I can tell, the code block below was meant to support EXPLODE expressions

Comment (barakalon) [reply sqlglot/expressions.py:849]: Let me see if I can clean this up..

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -846,10 +846,9 @@ def alias_column_names(self):
 
     @property
     def selects(self):
-        alias = self.args.get("alias")
+        if self.this and isinstance(self.this, Subqueryable):
+            return self.this.selects
 
-        if alias:
-            return alias.columns
         return []
```
Comment (GeorgeSittas) [inline sqlglot/expressions.py:852]: Maybe we can also do:
```python
return self.this.selects if isinstance(self.this, Subqueryable) else []
```

File: sqlglot/expressions.py [hunk 3]
```diff
@@ -920,7 +919,12 @@ def except_(self, expression, distinct=True, dialect=None, **opts):
 
 
 class UDTF(DerivedTable, Unionable):
-    pass
+    @property
+    def selects(self):
+        alias = self.args.get("alias")
+        if alias:
+            return alias.columns
+        return []
```
Comment (GeorgeSittas) [inline sqlglot/expressions.py:927]: One liner here as well?

File: sqlglot/expressions.py [hunk 4]
```diff
@@ -846,10 +846,9 @@ def alias_column_names(self):
 
     @property
     def selects(self):
-        alias = self.args.get("alias")
+        if self.this and isinstance(self.this, Subqueryable):
```
Comment (tobymao) [inline sqlglot/expressions.py:849]: don’t need the self.this

Comment (barakalon) [reply sqlglot/expressions.py]: doh

**Pull request merged**